### DISCUSSION TITLE
Wire @DgsDirective in orderly

### DISCUSSION THREAD

**Discussion by khacsinhcs** [Netflix/dgs-framework#1849] [Ideas]
```
    private fun findDirectives(applicationContext: ApplicationContext, runtimeWiringBuilder: RuntimeWiring.Builder) {
        applicationContext.getBeansWithAnnotation<DgsDirective>().values.forEach { directiveComponent ->
            val annotation = AopUtils.getTargetClass(directiveComponent).getAnnotation(DgsDirective::class.java)
            when (directiveComponent) {
                is SchemaDirectiveWiring ->
                    if (annotation.name.isNotBlank()) {
                        runtimeWiringBuilder.directive(annotation.name, directiveComponent)
                    } else {
                        runtimeWiringBuilder.directiveWiring(directiveComponent)
                    }
                else -> throw RuntimeException("Invalid @DgsDirective type: the class must implement graphql.schema.idl.SchemaDirectiveWiring")
            }
        }
    }
```
DGS currently strategy to wire SchemaDirectiveWiring is not keep the order. In my case, I need to wire my custom annotation in order. Can we sort by @Order provide by Spring

**Comment by srinivasankavitha** [comment]
What order are you expecting these in? It shouldn't technically matter for
what we are doing here, so curious to learn more about your use case.

**Reply by khacsinhcs** [reply]
We have created several custom GraphQL directives for validation and authorization purposes. Currently, we are looking to initiate APM (Application Performance Monitoring) transactions not only in Mutations/Queries but also within our custom DataFetcher. However, we've encountered a challenge: GraphQL-java automatically creates a DataFetcher for all fields. Our goal is to initiate transactions specifically in DgsDataFetchers. The issue arises because we need this new SchemaDirectiveWiring to execute first. This would enable us to check the type of DataFetcher before other SchemaDirectiveWirings potentially wrap the original DgsDataFetcher. This adjustment is crucial for us to ensure that APM transactions are started correctly and efficiently in our desired context.
